src/loader/lib/transpiler.js

```diff
@@ -87,26 +91,83 @@
 		}
 	}
 
-	function registerCustomPlugins(babel, plugins) {
-		var promises = [];
-		var self = this;
+	/**
+	 * Whether the plugin name is already registered in babel-standalone
+	 *
+	 * @param {{}} babel The babel object exported by babel-standalone
+	 * @param {string} pluginName The plugin name to be checked
+	 * @return {boolean}
+	 */
+	function isPluginRegistered(babel, pluginName) {
+		var availablePlugins = babel.availablePlugins || {};
+
+		return !!availablePlugins[pluginName];
+	}
 
-		function isAvailable(plugin) {
-			var availablePlugins = babel.availablePlugins || {};
+	/**
+	 * Register the custom plugins found in `babelOptions` object
+	 *
+	 * babel-standalone requires custom plugins to be registered in order to
+	 * be used, see https://github.com/babel/babel-standalone#customisation
+	 *
+	 * @param {{}} load The loader load
+	 * @param {{}} babel The babel object exported by babel-standalone
+	 * @param {array} plugins The babel plugins array
+	 * @return {Promise} Promise that resolves to the plugins array to be used
+	 *                   to transpile the load source code
+	 */
+	function registerCustomPlugins(load, babel, plugins) {
+		var self = this;
+		var promises = [];
+		var clonedPlugins = plugins.slice(0)
+
+		plugins.forEach(function(plugin, index) {
+			/**
+			 * `plugin` can be either a string or an array:
+			 *
+			 * ```
+			 * {
+			 *   plugins: [
+			 *     "pluginName",
+			 *     [ "pluginName", { ...pluginOptions }
+			 *   ]
+			 * }
+			 * ```
+			 */
+			var name = typeof plugin === "string" ? plugin : plugin[0];
+			var parent = self.configMain || "package.json!npm";
+
+			if (!isPluginRegistered(babel, name)) {
+				var pluginPromise = self.normalize(name, parent)
+					.then(function(name) {
+						// check if the load corresponds to a babel plugin,
+						// otherwise a loading plugin could try to load itself
+						if (load.name !== name) {
+							return self.import(name);
```

Comment:
I could be wrong but I feel like this needs to be `self["import"]` for IE9 maybe? At least that was the case with IE8, maybe should check if it's ok to use `.import` in IE9.